src/silicone/database_crunchers/rms_closest.py

```diff
@@ -130,43 +159,52 @@ def get_values_at_key_timepoints(idf, time_filter):
                     )
                 return to_return
 
-            lead_var_timeseries = get_values_at_key_timepoints(
+            lead_var_filt = get_values_at_key_timepoints(
                 lead_var, key_timepoints_filter_lead
-            ).timeseries()
+            )
+            lead_var_timeseries = lead_var_filt.timeseries()
             iamdf_lead_timeseries = get_values_at_key_timepoints(
                 iamdf_lead, key_timepoints_filter_iamdf
             ).timeseries()
 
             output_ts_list = []
-            for label, row in lead_var_timeseries.iterrows():
-                closest_ts = _select_closest(iamdf_lead_timeseries, row)
+            for _, (model, scenario) in (
+                lead_var_filt.data[["model", "scenario"]].drop_duplicates().iterrows()
+            ):
+                lead_var_mod_scen = lead_var_timeseries[
+                    (lead_var_timeseries.index.get_level_values("model") == model)
+                    & (
+                        lead_var_timeseries.index.get_level_values("scenario")
+                        == scenario
+                    )
+                ]
+                if len(lead_var_mod_scen) != len(variable_leaders):
+                    raise ValueError(
+                        "Insufficient variables are found to infill model {}, scenario {}".format(
```

Comment:
Maybe adding information about what variables were found in the error could be helpful?

Reply:
Yep, that'll probably be useful